### PR TITLE
[server] create a freemium team when no team exist

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
User's without a team, will get one with 500 credits free per month.

## Related Issue(s)

See #15786

## How to test

- Enable the migration (use `toa@` in your user's email)
 - verify your team has freemium. 
 - delete team
 - verify you automatically get a new one and that this one again has 500 free.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-payment
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/gitpod-db/src/user-to-team-migration-service.ts [hunk 1]
```diff
@@ -89,7 +89,18 @@ export class UserToTeamMigrationService {
...
-        log.info(ctx, "Migrated cost center data.", { teamId: team.id, result });
+        if (result.affectedRows === 0) {
+            const now = new Date();
+            const nextMonth = new Date();
+            nextMonth.setMonth(now.getMonth() + 1);
+            await conn.query(
+                "INSERT INTO d_b_cost_center (id, creationTime, spendingLimit, billingStrategy, billingCycleStart, nextBillingTime) VALUES (?,?,?,?,?,?)",
+                [newAttribution, now.toISOString(), 500, "other", now.toISOString(), nextMonth.toISOString()],
+            );
+            log.info(ctx, "Created cost center data.", { teamId: team.id, result });
```
Comment (easyCZ) [inline components/gitpod-db/src/user-to-team-migration-service.ts:100]: FWIW, the intent was that `usage` is the only component which will actually perform DB operations on the `cost_center` and `usage` records. With this, we're now departing from that. There's significant risk from multiple components interacting with the same tables as maintaining data quality becomes difficult.

What's the reasoning for abandoning that principle?

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:100]: > What's the reasoning for abandoning that principle?

The reason I took this approach is simplicity. The alternative would have been introducing much more non-local code (i.e. new service and API in usage component) to allow this, which would also introduce another dependency and with that would make the test infrastructure more complex. I see that there is a chance that a change to the underlying d_b_cost_center could go unnoticed and result in bad data, I believe the likely hood of introducing bugs by adding the complexity mentioned above is higher.

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:100]: FWIW I was looking into doing this in GetOrCreateCostCenter, but I'm missing the user information there.

Comment (easyCZ) [reply components/gitpod-db/src/user-to-team-migration-service.ts:100]: What stops us from adding that to the API surface?

If we always take the easier path to do something, the cost of that doesn't go away, it just accrues interest and comes back to us later on.

Comment (svenefftinge) [reply components/gitpod-db/src/user-to-team-migration-service.ts:100]: > If we always take the easier path to do something, the cost of that doesn't go away, it just accrues interest and comes back to us later on.

I believe it's not the easier path but the less costly (also long-term).

**Review by easyCZ — APPROVED** [review]

**Pull request merged**